clusterloader2/pkg/measurement/common/wait_for_controlled_pods.go

```diff
@@ -384,11 +403,35 @@ func (w *waitForControlledPodsRunningMeasurement) getObjectCountAndMaxVersion()
 			klog.Errorf("%s: retriving resource version error: %v", w, err)
 			continue
 		}
+		key, err := cache.DeletionHandlingMetaNamespaceKeyFunc(runtimeObj)
+		if err != nil {
+			klog.Errorf("%s: retrieving key error: %v", w, err)
+			continue
+		}
+		objectKeys = append(objectKeys, key)
 		desiredCount++
 		if version > maxResourceVersion {
 			maxResourceVersion = version
 		}
 	}
+	sort.Strings(objectKeys)
+
+	cond := func() (bool, error) {
+		currentKeys := w.informer.GetStore().ListKeys()
+		sort.Strings(currentKeys)
+		return reflect.DeepEqual(objectKeys, currentKeys), nil
```

Comment:
Hmm, there is a risk that if someone creates/deletes a resource that we're watching, between the List and the Poll in this method the condition will never be satisfied, e.g. 

1. We list: a,b,c
2. Someone adds d
3. Informer is at a,b,c,d
4. Condition will be always false

But I guess, it's fine. Given that a step can be either measurement or phase, CL2 itself will never modifying anything when this code.

Reply:
Yeah - that was my thinking. Added explicit comment.